advanced-usage.md

```diff
@@ -204,6 +204,82 @@ assert(r.elapsed <= 1); // Less than one second should have elapsed
 
 Setting the `Timeout` option sets the maximum allowed time the transfer operation can take. Since C++ Requests is built on top of libcurl, it's important to know what setting this `Timeout` does to the request. You can find more information about the specific libcurl option [here](http://curl.haxx.se/libcurl/c/CURLOPT_TIMEOUT_MS.html).
 
+## Setting Callbacks
+
+You can optionally set a callbacks for a request. Current there is support for read, write and progress callbacks.
+
+### ReadCallback
+
+This will be called every time curl is ready for data to be sent to the server.
+
+The layout of the read struct looks like this.
+
+```c++
+typedef struct {
+  void* userData = nullptr; // user data ptr.
+  void* buffer; // buffer to read into.
+  size_t realsize, size, nmeb; // actual size, chunk size, number of chunks.
+} ReadCallbackUser;
+```
+
+More information of this libcurl callback can be found [here](https://curl.haxx.se/libcurl/c/CURLOPT_READFUNCTION.html).
+
+### WriteCallback
+
+This will be called every time curl is ready to with a buffer to write. You can either choose to write directly to file, append the data to a buffer or save the data in a buffer until X size then write that chunk of memory to file.
+
+The layout of the write struct looks like this.
+
+```c++
+typedef struct {
+  void* userData = nullptr; // user data ptr.
+  void* buffer; // buffer containing downloaded data.
+  size_t realsize, size, nmeb; // actual size, chunk size, number of chunks.
+} WriteCallbackUser;
+```
+
+More information of this libcurl callback can be found [here](https://curl.haxx.se/libcurl/c/CURLOPT_WRITEFUNCTION.html).
+
+### ProgressCallback
+
+The progress callback will be called as often as libcurl can (often many times a second!).
+
+The layout of the progress struct looks like this.
+
+```c++
+typedef struct {
+  void* userData = nullptr; // pointer to your userdata
```

Comment:
For all struct comments:
Please start with a upper case letter if you end with a `.`